src/components/table-hoc/TableWithPagination.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState, IReduxActionsPayload} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IReduxAction, ReduxConnect} from '../../utils/ReduxUtils';
+import {turnOffLoading} from '../loading/LoadingActions';
+import {INavigationChildrenProps, INavigationOwnProps} from '../navigation/Navigation';
+import {NavigationConnected} from '../navigation/NavigationConnected';
+import {TableWithPaginationActions} from './actions/TablePaginationActions';
+import {ITableHOCOwnProps} from './TableHOC';
+
+export interface ITableWithPaginationConfig {
+    sliceData?: (data: any[], startingIndex: number, endingIndex: number) => any[];
+    isServer?: boolean;

```

Comment:
could you use an `IMaybeServerTable` interface and use it for all table components?